server/src/main/java/org/elasticsearch/synonyms/SynonymsManagementAPIService.java

```diff
@@ -278,6 +279,28 @@ public void getSynonymRule(String synonymSetId, String synonymRuleId, ActionList
         );
     }
 
+    public void deleteSynonymRule(
+        String synonymSetId,
+        String synonymRuleId,
+        ActionListener<SynonymsReloadResult<AcknowledgedResponse>> listener
+    ) {
+        checkSynonymSetExists(
+            synonymSetId,
+            listener.delegateFailure(
+                (l1, obj) -> client.prepareDelete(SYNONYMS_ALIAS_NAME, internalSynonymRuleId(synonymSetId, synonymRuleId))
+                    .setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE)
+                    .execute(new DelegatingIndexNotFoundActionListener<>(synonymSetId, l1, (l2, deleteResponse) -> {
```

Comment:
Why do we need an extra `DelegatingIndexNotFoundActionListener` here, doesn't `checkSynonymSetExists` already confirms that `synonymsSetId` exists?

Reply:
You're right - I was trying to cover for the case the system index does not yet exist, but `checkSynonymSetExists` does that. Thanks!